### ISSUE TITLE
Revise problematic stable labels

### ISSUE THREAD

**Issue by tkoeppe** [cplusplus/draft#1498]
Some of our stable labels are obnoxious (`string.op+`), debatable (`reverse.iter.op!=`), downright ridiculous (`istreambuf.iterator::op*`), or simply comical (`move.iter.op.+`).

We should review all our stable labels and change those that use unnecessarily special characters. Changing labels requires informing @Dani-Hub of the changes, so we should try and do all the work in one go.

We ought to be able to get a list of all labels from the "Cross references" glossary, but beware that certain special characters may have broken the indexing and thus don't even show up; check the glossary log separately.

**Comment by jensmaurer** [comment]
string.op+ is gone already, but needs fixing for string.op+=.
reverse.iter.* should have its sub-subsections removed (there are >5 subsections on a single page).
istreambuf.iterator::op* is already fixed.
move.iter.* should have its sub-subsections removed (there are >5 subsections on a single page).

**Comment by godbyk** [comment]
Here is the list of labels containing characters other than letters, numbers, underscore, period, or hyphen:

```
back.insert.iter.op*
back.insert.iter.op++
back.insert.iter.op=
front.insert.iter.op*
front.insert.iter.op++
front.insert.iter.op=
insert.iter.op*
insert.iter.op++
insert.iter.op=
ios::Init
ios::failure
ios::fmtflags
ios::iostate
ios::openmode
ios::seekdir
istream::sentry
move.iter.op.+
move.iter.op.+=
move.iter.op.-=
move.iter.op=
ostream::sentry
reverse.iter.op!=
reverse.iter.op+
reverse.iter.op++
reverse.iter.op+=
reverse.iter.op-=
reverse.iter.op<
reverse.iter.op<=
reverse.iter.op=
reverse.iter.op==
reverse.iter.op>
reverse.iter.op>=
string.op!=
string.op+
string.op+=
string.op<
string.op<=
string.op>
string.op>=
string.operator==
```

Once we've fixed the labels, we could add a CI test to check for these issues in the future.

**Comment by godbyk** [comment]
Here is the list of labels containing characters other than letters, numbers, underscore, period, or hyphen:

```
back.insert.iter.op*
back.insert.iter.op++
back.insert.iter.op=
front.insert.iter.op*
front.insert.iter.op++
front.insert.iter.op=
insert.iter.op*
insert.iter.op++
insert.iter.op=
ios::Init
ios::failure
ios::fmtflags
ios::iostate
ios::openmode
ios::seekdir
istream::sentry
move.iter.op.+
move.iter.op.+=
move.iter.op.-=
move.iter.op=
ostream::sentry
reverse.iter.op!=
reverse.iter.op+
reverse.iter.op++
reverse.iter.op+=
reverse.iter.op-=
reverse.iter.op<
reverse.iter.op<=
reverse.iter.op=
reverse.iter.op==
reverse.iter.op>
reverse.iter.op>=
string.op!=
string.op+
string.op+=
string.op<
string.op<=
string.op>
string.op>=
string.operator==
```

Once we've fixed the labels, we could add a CI test to check for these issues in the future.

**Comment by Quuxplusone** [comment]
If labels change, please inform @Dani-Hub *and @mclow* of the changes (as mclow mentioned in a comment on #1832; I mention it here for visibility).

Here is a shell one-liner (tested only with BSD `sed`, not GNU `sed`) for extracting all the stable names from the source code:

    git grep '^[^%]*\\rSec.\[.*\]{.*}' | grep -o '\[.*\]' | sort | sed 's/^.\(.*\).$/\1/'

You can pipe this through `grep '[^a-z0-9.]'` to get the current list of names that are IMHO troublesome.

Related to stable-name changes, I have also noticed that there is a very large number of conventions for naming sections that ought to be similarly named:

```
git grep 'rSec.\[.*\]{.*onstructor.*}' | grep -o '\[.*\]' > constructor-related-section-names.txt
grep -o '\.[^.]*\]' constructor-related-section-names.txt | sort | uniq -c

   3 .alloc]
   7 .cnstr]
  50 .cons]
   5 .const]
   1 .constr]
   2 .construct]
   2 .constructors]
  12 .ctor]
   1 .des]
   1 .functions]
   1 .init]
   1 .slice]
```

The surprising results here come from `constexpr.functions` and `class.inhctor.init` (false positives), `ostream.iterator.cons.des` and `cons.slice` (yuck!), and `queue.cons.alloc` etc. (debatable). The rest are not-horrible-but-unfortunate and would in an ideal world use a standard naming convention.

See also
```
git grep 'rSec.\[.*\]{.*estructor.*}' | grep -o '\[.*\]' > destructor-related-section-names.txt
git grep 'rSec.\[.*\]{.*\(ssignment\|operator=[^=]\).*}' | grep -o '\[.*\]' > assignment-related-section-names.txt
```

**Comment by jensmaurer** [comment]
Remaining offenders:

```ios::failure
ios::fmtflags
ios::Init
ios::iostate
ios::openmode
ios::seekdir
istream::sentry
ostream::sentry
string.op<=
string.op<
string.op>=
string.op>
string.op!=
string.op+=
string.op+
string.operator==```

**Comment by jensmaurer** [comment]
Remaining offenders:
```
class.mfct.non-static
ios::failure
ios::fmtflags
ios::Init
ios::iostate
ios::openmode
ios::seekdir
istream::sentry
ostream::sentry
```

**Comment by jensmaurer** [comment]
"class.mfct.non-static" is the only remaining questionable label (due to the hyphen), but that seems acceptable.

**Comment by Quuxplusone** [comment]
> `class.mfct.non-static` is the only remaining questionable label (due to the hyphen)

Peanut gallery asks: Why not fix that one by the same mechanism as we fixed all the other ones? (I suggest `class.mfct.nonstatic` as the obvious spelling.)

**Comment by jensmaurer** [comment]
@Quuxplusone, we have lots of stable labels with underscores in them, which look a lot more dreadful than the hyphen we have here. Those underscores are addressed in #2609.
